python/pyspark/ml/pipeline.py

```diff
@@ -89,9 +87,9 @@ def setParams(self, stages=None):
         setParams(self, stages=None)
         Sets params for Pipeline.
         """
-        if stages is None:
-            stages = []
         kwargs = self.setParams._input_kwargs
+        if stages is None:
+            kwargs['stages'] = []
```

Comment:
Yeah, I wouldn't just expect any Param to handle a value of `None`, but since it's in the function signature like that, it looks like it's a valid value.  Also, it seems like previously it was trying to check for this, so it might be good to keep.